app/controllers/staff/reports_state_controller.rb

```diff
@@ -47,10 +47,15 @@ def update
 
   private def change_report_state_to_active
     authorize report, :activate?
-    report.update!(state: :active)
-    report.create_activity key: "report.activated", owner: current_user
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/activate/complete"
+    if report.valid?
```

Comment:
Would it make sense to perform this check on all other state changes? This is the first change that's made one `change_report_state` method structurally different from any of the others, and as I'd considered refactoring all these methods down to a single implementation I'm curious about the need for them to differ.

Reply:
I've had a go a drying up refactor - to be honest, I think I prefer the more verbose way, let me know what you think and how I might improve this if I have missed the mark!?

Reply:
(I've just seen you've got something similar in